src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamReader.cs

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
```

Comment:
https://github.com/NuGet/NuGet.Client/pull/5529/files#r1415139352
> My gut reaction when I saw this is that it looks like you're copying parts of Utf8JsonReader from the .NET runtime repo. But after reading the PR description, and spending more time reading the PR, I see now that there are other reasons. However, once this code is merged, the PR description will no longer be quick to locate. Therefore, I think it's important to comment this class to explain why it exists, why we're not using STJ's Utf8JsonReader or JsonSerializer directly. - @zivkan 

I added a comment let me know what you think